zebra/zebra_rnh.c

```diff
@@ -153,20 +137,21 @@ struct rnh *zebra_add_rnh(struct prefix *p, vrf_id_t vrfid, enum rnh_type type,
 	struct route_table *table;
 	struct route_node *rn;
 	struct rnh *rnh = NULL;
-	char buf[PREFIX2STR_BUFFER];
 	afi_t afi = family2afi(p->family);
 
 	if (IS_ZEBRA_DEBUG_NHT) {
-		prefix2str(p, buf, sizeof(buf));
-		zlog_debug("%u: Add RNH %s type %s", vrfid, buf,
+		struct vrf *vrf = vrf_lookup_by_id(vrfid);
+
+		zlog_debug("%s(%u): Add RNH %pFX type %s", vrf->name, vrfid, p,
 			   rnh_type2str(type));
 	}
 	table = get_rnh_table(vrfid, afi, type);
 	if (!table) {
-		prefix2str(p, buf, sizeof(buf));
+		struct vrf *vrf = vrf_lookup_by_id(vrfid);
```

Comment:
This code block is entered if `get_rnh_table` returned NULL, which means that `vrf_lookup_by_id` will also return NULL here. `vrf` should be NULL-checked at least here, but I suppose the same thing may happen in other places as well.

Reply:
fixed

Reply:
looks good now